txpool/txpool.go

```diff
@@ -394,13 +394,10 @@ func (t *TxPool) addImpl(origin TxOrigin, tx *types.Transaction) error {
 // Since any discarded transaction should not affect the world state, the nextNonce should be reset to the value
 // it was set to before the transaction appeared.
 func (t *TxPool) DecreaseAccountNonce(tx *types.Transaction) {
-	mux := t.lockAccountQueue(tx.From, true)
-	defer mux.unlock()
+	wrapper := t.lockAccountQueue(tx.From, true)
+	defer wrapper.unlock()
 
-	wrapper, ok := t.accountQueues[tx.From]
-	if ok {
-		wrapper.accountQueue.nextNonce -= 1
-	}
+	wrapper.accountQueue.nextNonce -= 1
```

Comment:
This need to improved, nextnonce should be decreased to the nonce of the transaction , Check this [link](https://github.com/ethereum/go-ethereum/blob/721c5723c0e31c4d764c61ee99c505d18466b363/core/tx_pool.go#L1021) for reference.

Reply:
Got clarity from milos

Reply:
@munna0908 @zivkovicmilos 's PR does the change where the nextNonce is always in sync with the state. I don't think your comment is relevant in this case.